pkg/manager/watch_endpoints.go

```diff
@@ -82,41 +82,23 @@ func (ep *endpointsProvider) getAllEndpoints() ([]string, error) {
-			log.Debugf("[%s] %v, shortname will not be used", ep.label, shortnameErr)
-		} else {
-			log.Errorf("[%s] %v", ep.label, shortnameErr)
-		}
-	}
-
-	for subset := range ep.endpoints.Subsets {
-		for address := range ep.endpoints.Subsets[subset].Addresses {
-			// 1. Compare the hostname on the endpoint to the hostname
-			// 2. Compare the nodename on the endpoint to the hostname
-			// 3. Drop the FQDN to a shortname and compare to the nodename on the endpoint
-
-			// 1. Compare the Hostname first (should be FQDN)
-			log.Debugf("[%s] processing endpoint [%s]", ep.label, ep.endpoints.Subsets[subset].Addresses[address].IP)
-			if id == ep.endpoints.Subsets[subset].Addresses[address].Hostname {
-				log.Debugf("[%s] found local endpoint - address: %s, hostname: %s",
-					ep.label, ep.endpoints.Subsets[subset].Addresses[address].IP, ep.endpoints.Subsets[subset].Addresses[address].Hostname)
-				localEndpoints = append(localEndpoints, ep.endpoints.Subsets[subset].Addresses[address].IP)
-			} else {
-				// 2. Compare the Nodename (from testing could be FQDN or short)
-				if ep.endpoints.Subsets[subset].Addresses[address].NodeName != nil {
-					if id == *ep.endpoints.Subsets[subset].Addresses[address].NodeName {
-						log.Debugf("[%s] found local endpoint - address: %s, hostname: %s, node: %s",
-							ep.label, ep.endpoints.Subsets[subset].Addresses[address].IP, ep.endpoints.Subsets[subset].Addresses[address].Hostname,
-							*ep.endpoints.Subsets[subset].Addresses[address].NodeName)
-						localEndpoints = append(localEndpoints, ep.endpoints.Subsets[subset].Addresses[address].IP)
-					} else if shortnameErr == nil && shortname == *ep.endpoints.Subsets[subset].Addresses[address].NodeName {
-						log.Debugf("[%s] found local endpoint -  address: %s, shortname: %s, node: %s",
-							ep.label, ep.endpoints.Subsets[subset].Addresses[address].IP, shortname, *ep.endpoints.Subsets[subset].Addresses[address].NodeName)
-						localEndpoints = append(localEndpoints, ep.endpoints.Subsets[subset].Addresses[address].IP)
-					}
-				}
+	for subsetIndex := range ep.endpoints.Subsets {
+		subset := ep.endpoints.Subsets[subsetIndex]
+		for addressIndex := range subset.Addresses {
+			address := subset.Addresses[addressIndex]
+			log.Debugf("[%s] processing endpoint [%s]", ep.label, address.IP)
+
+			// 1. Compare the Nodename
+			if address.NodeName != nil && id == *address.NodeName {
+				log.Debugf("[%s] found local endpoint - address: %s, hostname: %s, node: %s", ep.label, address.IP, address.Hostname, *address.NodeName)
+				localEndpoints = append(localEndpoints, address.IP)
+				continue
+			}
+			// 2. Compare the Hostname (only useful if node name is not available)
+			if id == address.Hostname {
+				log.Debugf("[%s] found local endpoint - address: %s, hostname: %s", ep.label, address.IP, address.Hostname)
+				localEndpoints = append(localEndpoints, address.IP)
+				continue
```

Comment:
Here and in `watch_endpointslices.go` I removed short hostname-to-nodename matching because k8s is already providing the correct value.
First I check `address.NodeName` because documentation states "This can be used to determine endpoints local to a node".
`address.Hostname` doesn't seem relevant to me, but I left it here as a second check, just in case this is relevant in some setups. Maybe it should also be removed.